src/Extension/CoreExtension.php

```diff
@@ -1573,6 +1573,10 @@ function twig_array_filter($array, $arrow)
         return array_filter($array, $arrow, \ARRAY_FILTER_USE_BOTH);
     }
 
+    if (!$array instanceof \Traversable) {
+        throw new RuntimeError(sprintf('An array filter only works with arrays or "Traversable", got "%s" as first argument.', \gettype($array)));
```

Comment:
```suggestion
        throw new RuntimeError(sprintf('The filter filter only works with arrays or "Traversable", got "%s" as first argument.', \gettype($array)));
```
(or maybe
```suggestion
        throw new RuntimeError(sprintf('The "filter" filter only works with arrays or "Traversable", got "%s" as first argument.', \gettype($array)));
```
to mitigate the repetition)